tools: Refactor trap tool

thapav opened the pull request:
Remove kernel symbols added twice.
    Remove elf file argument to ramdumpParser.py
    Add elf file extension parsing to ramdumpParser.py
    Rectify HowToUseTrap.md log formatting.

Signed-off-by: Vidisha <thapa.v@samsung.com>

tizen-build commented:
Target : [2af2222196bbf59fc9d6f8ffeb787f875c08bb91] - Code Rule Check OK.

sunghan-chang commented:
Could you write what you refactor in the commit description?

abhishek-samsung commented on tools/trap/HowToUseTrap.md:
```diff
@@ -303,6 +303,7 @@ PC_value	 Symbol_address	  Symbol_name	File_name
 App Crash point is as follows:
 [ Caller - return address (LR) - of the function which has caused the crash ]
 
+App name          : app2
```
Please change the usage. Now we need to pass the kernel elf file as an argument compulsorily. Currently, we can use any elf file according to the instructions above.
Example line 283 : $ python3 ramdumpParser.py -e ../../build/output/bin/app2_dbg -t ./log_file

thapav replied:
Done

tizen-build commented:
Target : [cf2b2a52429bb586c9da068db18b0cf33e871b06] - Code Rule Check OK.

tizen-build commented:
Target : [ec89a1c31a35a2d7c0969b22527cc5ae747ecd23] - Code Rule Check OK.

sunghan-chang commented on tools/trap/HowToUseTrap.md:
```diff
@@ -280,7 +280,7 @@ cd $TIZENRT_BASEDIR/tools/trap/
 
     ex)
-    $ python3 ramdumpParser.py -e ../../build/output/bin/app2_dbg -t ./log_file
+    $ python3 ramdumpParser.py -e ../../build/output/bin/tinyara -t ./log_file
```
Regardless of crash point, can we use `tinyara` only?
If it is, why do we need to pass only this?

thapav replied:
Yes only tinyara can be used.
For app crash point detection we calculate app*_dbg based on the applications loaded for the configuration. (Line 592)

sunghan-chang replied:
Then, why do we need to pass -e option even app can be done automatically?

sunghan-chang replied:
If we need to pass the pass of bin folder for the location of elfs, then how about removing `tinyara` like `-e ../../build/output/bin`? If we can do that, then we had better change the option from -e to -l (?).

thapav replied:
Only apps have an elf naming convention like <app_name>_dbg
For kernel elf varies > tinyara/tinyara.axf 
Hence using this option.

sunghan-chang replied:
Kernel also has naming convention. Please find `EXEEXT` in os/Make.defs.
naming convention of tinyara
https://github.com/Samsung/TizenRT/blob/master/os/Makefile.unix#L228

sunghan-chang replied:
Only rtl8721csm has the EXEEXT value as axf so that tinyara.axf is used. Other boards has nothing for that. That's why tinyara is used for other boards.

sunghan-chang replied:
Even we change not to pass kernel elf, we need to pass the location because we can't guarantee this script can be executed at the same location.
Could you let me know your opinion for giving kernel elf and for giving location by user?

sunghan-chang commented on tools/trap/HowToUseTrap.md:
````diff
@@ -373,7 +375,7 @@ PC_value	 Symbol_address	  Symbol_name	File_name
 ```
 ex)
-$python3 ramdumpParser.py -e ../../build/output/bin/common_dbg -t ./log_file
+$python3 ramdumpParser.py -e ../../build/output/bin/tinyara -t ./log_file
````
same as previous

sunghan-chang reviewed (commented):
Could you check line 47 on md file, `## How to upload RAMDUMP-or-UserfsDUMP`?
Let's remove `-`s.

thapav commented:
@sunghan-chang Updated the PR. Kindly check
As per our discussion,
I have removed elf argument mandate from the script. The extension is also extracted from the Make.defs.
Log file naming convention is also flexible now.

tizen-build commented:
Target : [2476aed0b29de71a3d9c4fafefcfdc3d5fb423c6] - Code Rule Check OK.

tizen-build commented:
Target : [9337130381f10a9baf578c7992b4a66162398d04] - Code Rule Check OK.

sunghan-chang reviewed (commented):
1. change DumpParser -> TRAP
2. please check line 272. The explanation says run the script but box has changing the directory only.

thapav commented:
> 

modified

tizen-build commented:
Target : [b049975bae9b53b0534fec396a6b2ded937d1dc1] - Code Rule Check OK.

sunghan-chang commented on tools/trap/HowToUseTrap.md:
````diff
@@ -262,32 +262,36 @@ Ramdump received successfully..!
 cd $TIZENRT_BASEDIR/tools/trap/
 ```
-2. See the Output
+2. Run Ramdump Parser Script and see the Output
````
There is no guide to run ramdump parser script. Please add that like 1st step.

tizen-build commented:
Target : [af1e9c2857ba18a7e15d712ca38a417417646e11] - Code Rule Check OK.

tizen-build commented:
Target : [0ab267fc09cde986cf309c63f7715efedea8f83c] - Code Rule Check OK.

sunghan-chang commented on tools/trap/HowToUseTrap.md:
````diff
@@ -262,32 +262,91 @@ Ramdump received successfully..!
 cd $TIZENRT_BASEDIR/tools/trap/
 ```
-2. See the Output
+2. Copy crash logs  
+    First copy the crash logs to a file in tools/trap/`<log_file>`
````
Do we need this 2nd step even this tab is `To get call stack using RAM dump`? If it is, why do you guide two different tabs with the same content? And what can we do with ramdump? I mean why do we need ramdump?

thapav replied:
I think we can merge the two under one menu
"To get call stack"
Ramdump is used to retrieve call stack for scenarios when crash logs are not accessible and for the gui_ramdumpParser.py script.

sunghan-chang replied:
Are you still modifying? There is no change on this.

tizen-build commented:
Target : [f10ea91152d5fa189057e9b803c117e6ed78f50d] - Code Rule Check OK.

sunghan-chang commented on tools/debug/debugsymbolviewer.py:
```diff
@@ -161,7 +161,7 @@ def parse_file(log_file):
 					if (is_text_address(hex(stack_addr))):
 						#If yes, print it's corresponding symbol
                                                 stack_addr = stack_addr - int(offset,16)
-						print_symbol(stack_addr)
+                                                print_symbol(stack_addr)
```
Except those two lines, you used tabs for indentation. I don't know this change is necessary.

tizen-build commented:
Target : [855b3ecbf331315a150cec0fba59dbf63d3b48c3] - Code Rule Check OK.
